src/resolvers.ts

```diff
@@ -22,17 +22,34 @@ export const resolveConfigPath = async (path: Arrayable<string>) => (
  * @param srcDir
  * @returns array of resolved content globs
  */
-export const resolveContentPaths = (srcDir: string) => ([
-  `${srcDir}/components/**/*.{vue,js,ts}`,
-  `${srcDir}/layouts/**/*.vue`,
-  `${srcDir}/pages/**/*.vue`,
-  `${srcDir}/composables/**/*.{js,ts}`,
-  `${srcDir}/plugins/**/*.{js,ts}`,
-  `${srcDir}/utils/**/*.{js,ts}`,
-  `${srcDir}/{App,app}.{js,ts,vue}`,
-  `${srcDir}/{Error,error}.{js,ts,vue}`,
-  `${srcDir}/app.config.{js,ts}`
-])
+export const resolveContentPaths = (srcDir: string, nuxt = useNuxt()) => {
+  const r = (p: string) => p.startsWith(srcDir) ? p : `${srcDir}/${p}`
+  const extensionFormat = (s: string[]) => s.length > 1 ? `.{${s.join(',')}}` : `.${s.join('') || 'vue'}`
+
+  const defaultExtensions = extensionFormat(['js', 'ts', 'mjs'])
+  const sfcExtensions = extensionFormat(nuxt.options.extensions.map(e => e.replace(/^\.*/, '')))
+
+  return [
+    ...(() => {
+      if (nuxt.options.components) {
+        return (Array.isArray(nuxt.options.components) ? nuxt.options.components : typeof nuxt.options.components === 'boolean' ? ['components'] : nuxt.options.components.dirs).map(d => `${resolveAlias(typeof d === 'string' ? d : d.path)}/**/*${sfcExtensions}`)
+      }
+      return []
+    })(),
+
+    r(`${nuxt.options.dir.layouts}/**/*${sfcExtensions}`),
+    ...(nuxt.options.pages ? [r(`${nuxt.options.dir.pages}/**/*${sfcExtensions}`)] : []),
+
+    r(`${nuxt.options.dir.plugins}/**/*${defaultExtensions}`),
+    r(`${nuxt.options.dir.modules}/**/*${defaultExtensions}`),
+    ...nuxt.options.modulesDir.map(m => r(`${m}/**/*${defaultExtensions}`)),
+    ...(nuxt.options.imports.dirs || []).map(d => r(`${d}/**/*${defaultExtensions}`)),
```

Comment:
Suppose I'll move my comment to the new PR as well. `imports.dirs` doesn't end up including composables and utils as they're added by nuxt regardless of what the config value is.

Reply:
How do you feel about this:

```ts

  const importDirs = [...(nuxt.options.imports.dirs || [])]
  const [composablesDir, utilsDir] = [resolve(srcDir, 'composables'), resolve(srcDir, 'utils')]

  if (!importDirs.includes(composablesDir)) importDirs.push(composablesDir)
  if (!importDirs.includes(utilsDir)) importDirs.push(utilsDir)

  // later..
 ...importDirs.map(d => r(`${d}/**/*${defaultExtensions}`)),
```

Reply:
(committed)

Reply:
`imports.dirs` should probably be mapped to an absolute path before attempting to deduplicate. Might also want to use a Set instead.

Something like this:
```js
const importDirs = new Set([...(nuxt.options.imports.dirs || [])].map(it => resolve(it)))
importDirs.add(resolve(srcDir, 'composables'))
importDirs.add(resolve(srcDir, 'utils'))
```